compiler/rustc_typeck/src/check/check.rs

```diff
@@ -1140,16 +1140,21 @@ pub fn check_simd(tcx: TyCtxt<'_>, sp: Span, def_id: LocalDefId) {
             struct_span_err!(tcx.sess, sp, E0075, "SIMD vector cannot be empty").emit();
             return;
         }
+        let param_env = tcx.param_env(def.did());
         let e = fields[0].ty(tcx, substs);
-        if !fields.iter().all(|f| f.ty(tcx, substs) == e) {
+        let e = tcx.normalize_erasing_regions(param_env, e);
```

Comment:
this (and below) should use `fcx.normalize_associated_types_in` I'm pretty sure. though I am suspicious why this normalization call is necessary to begin with.

also, we should not need to get the `param_env` of another item to do these normalizations, I think?

Reply:
yea, all new normalization calls should be documented with a comment explaining the reason for them to exist

Reply:
Should I write `// Needed because of TyAlias`?

Reply:
@compiler-errors To build a `fcx`, I need a `Inherited` that I can't seem to get at this point. Also, we are checking a struct there, so is `FnCtxt` really what we want to use?

Reply:
> Should I write `// Needed for TyAlias`?

No, that comment explains nothing. What I meant is a comment explaining what kind of code causes this path to be hit. Potentially referencing a test that causes it to be hit.